examples/CSVImporter/CSVImporter.go

```diff
@@ -0,0 +1,254 @@
+            1 : &currentUser.LastName,
+            2 : &currentUser.UserName,
+            3 : &currentUser.Email,
+            4 : &currentUser.Uid,
+            5 : &currentUser.Gid,
+            // "Sudo" boolean will be handled separately, so no 6
+            7 : &currentUser.Password,
+            8 : &currentHost,
+            9 : &currentTag,
+        }
+
+        // (Re)Fetch all users in JumpCloud (pickup any newly added users)
+        userList, err := jc.GetSystemUsers(false)
+
+        if err != nil {
+            fmt.Printf("Could not read system users, err='%s'\n", err)
+            return
+        }
+
+        // Read next record from CSV file
+    	record, err := reader.Read()
+
+        // Exit loop at the end of file or on error
+    	if err == io.EOF {
+    		fmt.Println("EOF")
+    		break
+    	} else if err != nil {
+    		fmt.Println(err)
+    		return
+    	}
+
+        recordCount = recordCount + 1
+
+    	// Parse the record just read into our work vars
+    	for i, element := range record {
+            // Handle variable fields separately
+            if i > 9 {
+                break
+            }
+
+            // Special case for sole boolean to be parsed
+            if i == 6 {
+                currentUser.Sudo = jcapi.GetTrueOrFalse(element)
+            } else {
+                // Default case is to move the string into the var
+                *fieldMap[i] = element
+            }
+    	}
+
+        // The administrators list is optional, and variable.  Using a slice
+        // will pick it up if it exists without causing errors otherwise.
+        // Map any names found to their ID values for later use.
+        adminsSlice := record[10:]
+
+        for _, tempAdmin := range adminsSlice {
+            currentAdmins[tempAdmin] = GetUserIdFromUserName(userList, tempAdmin)
+        }
+
+        // Determine operation to perform based on whether the current user
+        // is already in JumpCloud...
+        var opCode jcapi.JCOp
+        currentUserId := GetUserIdFromUserName(userList, currentUser.UserName)
+
+        if currentUserId != "" {
+            opCode = jcapi.Update
+            currentUser.Id = currentUserId
+        } else {
+            opCode = jcapi.Insert
+        }
+
+        // Perform the requested operation on the current user and report results
+        currentUserId, err = jc.AddUpdateUser(opCode, currentUser)
+
+        if err != nil {
+            fmt.Printf("Could not process user '%s', err='%s'", currentUser.ToString(), err)
+            continue
+        } else {
+            fmt.Printf("Processed user ID '%s'\n", currentUserId)
+        }
+
+        // Create/associate JumpCloud tags for the host and user...
+        if currentHost != "" {
+            // Determine if the host specified is defined in JumpCloud
+            var currentJCSystem jcapi.JCSystem
+
+            tempSysList, err := jc.GetSystemByHostName(currentHost, true)
+
+            if len(tempSysList) > 0 {
```

Comment:
We should probably error if we find more than one host that matches the host name. We won't have any idea which host to associate the user to. Hostnames are not guaranteed unique within an account, especially since they are not FQDNs.